azure-pipelines.yml

```diff
@@ -8,108 +8,100 @@ pr:
-    Get-ComputerInfo |
-      select WindowsProductName, WindowsVersion, OsHardwareAbstractionLayer |
-      Out-String |
-      Write-Host
-    #
-    # Azure
-    #
-    $assetTag = Get-WmiObject -class Win32_SystemEnclosure -namespace root\CIMV2 |
-      Select -ExpandProperty SMBIOSAssetTag
+    $gemfile = Join-Path -Path (Get-Location) -ChildPath 'docker/Gemfile'
+    $gempath = Join-Path -Path (Get-Location) -ChildPath 'docker/.bundle/gems'
+    bundle config --local gemfile $gemfile
+    bundle config --local path $gempath
+    bundle install
+  displayName: Fetch Dependencies
+  name: fetch_deps
 
-    # only Azure VMs have this hard-coded DMI value
-    if ($assetTag -eq '7783-7084-3265-9085-8269-3286-77')
-    {
-      Write-Host "`n`n$line`nAzure`n$line`n"
-      Invoke-RestMethod -Headers @{'Metadata'='true'} -URI http://169.254.169.254/metadata/instance?api-version=2017-12-01 -Method Get |
-        ConvertTo-Json -Depth 10 |
-        Write-Host
-    }
-    #
-    # Docker
-    #
-    Write-Host "`n`n$line`nDocker`n$line`n"
-    docker version
-    docker images
-    docker info
-    sc.exe qc docker
-    #
-    # Ruby
-    #
-    Write-Host "`n`n$line`nRuby`n$line`n"
-    ruby --version
-    gem --version
-    gem env
-    bundle --version
-    #
-    # Environment
-    #
-    Write-Host "`n`n$line`nEnvironment`n$line`n"
-    Get-ChildItem Env: | % { Write-Host "$($_.Key): $($_.Value)"  }
+- powershell: |
+    . "$(bundle show pupperware)/ci/build.ps1"
+    Write-HostDiagnostics
   displayName: Diagnostic Host Information
   name: hostinfo
+
 - powershell: |
-    . ./docker/ci/build.ps1
-    Lint-Dockerfile -Path ./docker/puppet-agent-ubuntu/Dockerfile
-    Lint-Dockerfile -Path ./docker/puppet-agent-alpine/Dockerfile
-  displayName: Lint Ubuntu and Alpine Dockerfiles
-  name: lint_dockerfiles
+    . "$(bundle show pupperware)/ci/build.ps1"
+    Lint-Dockerfile -Name $ENV:CONTAINER_NAME2 -Ignore ($ENV:LINT_IGNORES -split ' ')
+  displayName: Lint $(CONTAINER_NAME2) Dockerfile
+  name: lint_alpine_dockerfile
+
 - powershell: |
-    . ./docker/ci/build.ps1
-    Build-Container -Namespace $ENV:NAMESPACE -Base 'ubuntu'
-  displayName: Build Ubuntu Agent Container
-  name: build_puppetagent_ubuntu_container
+    . "$(bundle show pupperware)/ci/build.ps1"
+    Lint-Dockerfile -Name $ENV:CONTAINER_NAME -Ignore ($ENV:LINT_IGNORES -split ' ')
+  displayName: Lint $(CONTAINER_NAME) Dockerfile
+  name: lint_dockerfile
+
 - powershell: |
     . ./docker/ci/build.ps1
-    Invoke-ContainerTest -Name puppet-agent-ubuntu -Namespace $ENV:NAMESPACE
-  displayName: Test Puppet Agent Ubuntu
-  name: test_puppet_agent_ubuntu
-- task: PublishTestResults@2
-  displayName: Publish Puppet Agent Ubuntu test results
-  inputs:
-    testResultsFormat: 'JUnit'
-    testResultsFiles: 'docker/TEST-*.xml'
-    #mergeTestResults: false # Optional
-    testRunTitle: Puppet Agent Ubuntu Test Results
+    Build-Container -Namespace $ENV:NAMESPACE -Base 'ubuntu'
```

Comment:
I don't recognize this `-Base` option ?

Reply:
Oh it's not coming from the shared one, it's coming from the local one

Reply:
That's a leftover from the existing `Build-Container` helper that I left alone for now (rather than from the shared one)